test/src/unit-enumerations.cc

```diff
@@ -1550,6 +1558,44 @@ TEST_CASE_METHOD(
   REQUIRE(node2->use_enumeration() == true);
 }
 
+TEST_CASE_METHOD(
+    EnumerationFx,
+    "Cap'N Proto - Basic Array v2 Serialization",
+    "[enumeration][capnp][serialization][v2][array]") {
+  auto client_side = GENERATE(true, false);
+  auto ser_type = GENERATE(SerializationType::CAPNP, SerializationType::JSON);
+  auto do_load = GENERATE(std::string("true"), std::string("false"));
+
+  create_array();
+
+  Config cfg;
+  throw_if_not_ok(cfg.set("rest.use_refactored_array_open", "true"));
+  throw_if_not_ok(cfg.set("rest.load_enumerations_on_array_open", do_load));
+  Context ctx(cfg);
+
+  auto a1 = make_shared<Array>(HERE(), uri_, ctx.storage_manager());
+  throw_if_not_ok(
+      a1->open(QueryType::READ, EncryptionType::NO_ENCRYPTION, nullptr, 0));
+  REQUIRE(a1->serialize_enumerations() == (do_load == "true" ? true : false));
+  REQUIRE(
+      a1->array_schema_latest_ptr()->get_loaded_enumeration_names().size() ==
+      0);
+
+  auto a2 = make_shared<Array>(HERE(), uri_, ctx.storage_manager());
+
+  ser_des_array(ctx, a1.get(), a2.get(), client_side, ser_type);
```

Comment:
Ideally this type of tests should validate a behaviour based on rountrip rest interaction
ser_something(client_side = true), des_something(client_side=false), ser_something(client_side=false), des_something(client_side=true).
The behavior we want to test is that opening array in v2 mode makes enumeration info available on the client side, that implies a round trip rest interaction.

Reply:
Absolutely on adding more REST tests. The issue is that we can't actually commit those tests until after TileDB-Cloud-REST has been updated to use the code we want to test. For instance, the work in #4288 has a companion branch of the REST tests I wrote (and ran locally to prove they work). But I can't PR that until after #4288 is merged and tiledb-cloud-rest has a branch picking up the new code to be tested.

These tests just show that the actual code paths I've added are exercised even if it is a bit superficial in that we just assume the config getting passed through REST land happens correctly.